src/lib/Microsoft.Health.Fhir.Ingest.Template/CalculatedFunction/CalculatedFunctionContentTemplate.cs

```diff
@@ -0,0 +1,210 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using EnsureThat;
+using Microsoft.Health.Fhir.Ingest.Data;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.Ingest.Template.CalculatedFunction
+{
+    public class CalculatedFunctionContentTemplate : IContentTemplate
+    {
+        public const string MatchedToken = "matchedToken";
+
+        [JsonProperty(Required = Required.Always)]
+        public virtual string TypeName { get; set; }
+
+        [JsonProperty(Required = Required.Always)]
+        public virtual string TypeMatchExpression { get; set; }
+
+        public virtual ExpressionLanguage? TypeMatchExpressionLanguage { get; set; }
+
```

Comment:
The reason for a separate `XxxMatchExpression` and `XxxMatchExpressionLanguage` fields was to make the Swagger definition for this new template easier to write. I do think it would be easier to make the `XxxMatchExpression` an object that contains the expression and language. Doing this would reduce the duplication that we see here. So if we can produce Swagger that would support that I can make that change. Doesn't need to be in this PR, we can do it in a subsequent one.